[BUG] OIIO no longer builds against split IlmBase and OpenEXR builds

Historically we've always released IlmBase and OpenEXR to separate install trees. However, it seems that the change to support OpenEXR 3 (ab3f04204ba228ca778044c7a4e84d157695e4aa) has broken this.

I've tested trying to build 2.2.13 against split releases of OpenEXR 2.3, 2.4, and 2.5, all with the same (failing) result. I've also confirmed that OIIO *will* build against a combined IlmBase/OpenEXR 2.5 install.

Here is the CMake command I'm using in both cases:
```
    cmake
    -G "Unix Makefiles"
    -D CMAKE_INSTALL_PREFIX=${REZ_INSTALL_PREFIX}
    -D CMAKE_BUILD_TYPE=${REZ_RELEASE_TYPE}
    -D CMAKE_EXE_LINKER_FLAGS=-Wl,--enable-new-dtags
    -D CMAKE_SHARED_LINKER_FLAGS=-Wl,--enable-new-dtags
    -D CMAKE_CXX_STANDARD=11
    -D BUILD_SHARED_LIBS=ON
    -D LINKSTATIC=ON
    -D EMBEDPLUGINS=ON
    -D GLIBCXX_USE_CXX11_ABI=0
    -D USE_SIMD=sse4.2
    -D TIFF_INCLUDE_DIR=${tiff_INCLUDE_DIRS}
    -D TIFF_LIBRARY=${tiff_libtiff_STATIC_LIBRARY}
    -D PNG_PNG_INCLUDE_DIR=${png_INCLUDE_DIRS}
    -D PNG_LIBRARY=${png_libpng15_STATIC_LIBRARY}
    -D JPEG_INCLUDE_DIR=${jpeg_INCLUDE_DIRS}
    -D JPEG_LIBRARY=${jpeg_libjpeg_STATIC_LIBRARY}
    -D CMAKE_LIBRARY_PATH=${lcms_LIBRARY_DIRS}
    -D BUILD_MISSING_FMT=OFF
    -D BUILD_MISSING_PYBIND11=OFF
    -D Pybind11_ROOT=${REZ_PYBIND11_ROOT}
    -D BUILD_MISSING_ROBINMAP=OFF
    -D Robinmap_ROOT=${REZ_ROBIN_MAP_ROOT}
    -D USE_LIBRAW=ON
    -D LibRaw_ROOT=${REZ_LIBRAW_ROOT}
    -D USE_PYTHON=ON
    -D Python_ADDITIONAL_VERSIONS=${PYTHON_MAJOR_VERSION}.${PYTHON_MINOR_VERSION}
    -D USE_OPENCOLORIO=OFF
    -D USE_DCMTK=OFF
    -D USE_FFMPEG=OFF
    -D USE_FIELD3D=OFF
    -D USE_GIF=OFF
    -D USE_HDF5=OFF
    -D USE_LIBHEIF=OFF
    -D USE_OPENCV=OFF
    -D USE_OPENJPEG=OFF
    -D USE_OPENVDB=OFF
    -D USE_PTEX=OFF
    -D USE_QT=OFF
    -D USE_WEBP=OFF
    -D STOP_ON_WARNING=OFF
    -D VERBOSE=ON
```

**Relevant output for the broken build (against the split-release of IlmBase/OpenEXR):**

Find information for OpenEXR:
```
-- Found OpenEXR: /luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/include  
-- Found OpenEXR 2.5.5 
--     OPENEXR_INCLUDES = /luma/soft/rez/packages/openexr/2.5.5/platform-linux/gcc-4.8.3/include
--     OPENEXR_INCLUDE_DIR = /luma/soft/rez/packages/openexr/2.5.5/platform-linux/gcc-4.8.3/include
--     OPENEXR_LIBRARIES = /luma/soft/rez/packages/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libIlmImf-2_5.a;/luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/lib64/libImath-2_5.a;/luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/lib64/libIex-2_5.a;/luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/lib64/libHalf-2_5.a;/luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/lib64/libIlmThread-2_5.a;-lpthread;/usr/lib64/libz.so
--     IMATH_INCLUDES = /luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/include
```

First point of failure:
```
[  0%] Building CXX object src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o
cd /mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/src/libutil && ccache /usr/bin/g++  -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_BOOST_REGEX -DUSE_FREETYPE=1 -DUSE_JPEG_TURBO=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include/OpenImageIO -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/src/include -I/mnt/bonus/dev/rez-build/openimageio/dev/src/src/include -I/luma/soft/rez/packages/fmt/6.1.2/platform-linux/include -I/luma/dev/ruschn/rez-release/openimageio/2.2.13.0/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include -isystem /luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/include -isystem /luma/soft/rez/packages/boost/1.61.0/platform-linux/gcc-4.8.3/python-2.7/include -isystem /luma/soft/rez/packages/ilmbase/2.5.5/platform-linux/gcc-4.8.3/include/OpenEXR  -O3 -DNDEBUG -fPIC   -Wall -Wno-unused-local-typedefs -Wno-unused-result -fno-math-errno -msse4.2 -std=c++11 -o CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -c /mnt/bonus/dev/rez-build/openimageio/dev/src/src/libutil/argparse.cpp
In file included from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/fmath.h:40:0,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/hash.h:31,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/strutil.h:24,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/typedesc.h:29,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/attrdelegate.h:13,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/paramlist.h:17,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/argparse.h:20,
                 from /mnt/bonus/dev/rez-build/openimageio/dev/src/src/libutil/argparse.cpp:16:
/mnt/bonus/dev/rez-build/openimageio/dev/src/src/include/OpenImageIO/Imath.h:13:35: fatal error: OpenEXR/OpenEXRConfig.h: No such file or directory
 #include <OpenEXR/OpenEXRConfig.h>
```

The basic issue is that `OpenEXR/OpenEXRConfig.h` is installed as part of OpenEXR (not IlmBase), but the compile command is only including the IlmBase include dirs.

**Relevant output for the working build (against combined IlmBase/OpenEXR install):**

Find information for OpenEXR:
```
-- Found OpenEXR 2.5.5 
--     OPENEXR_INCLUDES = /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/include
--     OPENEXR_INCLUDE_DIR = /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/include
--     OPENEXR_LIBRARIES = /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libIlmImf-2_5.a;/luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libImath-2_5.a;/luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libIex-2_5.a;/luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libHalf-2_5.a;/luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/lib64/libIlmThread-2_5.a;-lpthread;/usr/lib64/libz.so
--     IMATH_INCLUDES = /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/include
```

Working compile command for same object as above:
```
[  0%] Building CXX object src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o
cd /mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/src/libutil && ccache /usr/bin/g++  -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_BOOST_REGEX -DUSE_FREETYPE=1 -DUSE_JPEG_TURBO=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include/OpenImageIO -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include -I/mnt/bonus/dev/rez-build/openimageio/dev/build/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/src/include -I/mnt/bonus/dev/rez-build/openimageio/dev/src/src/include -I/luma/soft/rez/packages/fmt/6.1.2/platform-linux/include -I/luma/dev/ruschn/rez-release/openimageio/2.2.13.0/platform-linux/gcc-4.8.3/boost==1.61.0/python-2.7/!nuke/include -isystem /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/include -isystem /luma/soft/rez/packages/boost/1.61.0/platform-linux/gcc-4.8.3/python-2.7/include -isystem /luma/dev/ruschn/rez-release/openexr/2.5.5/platform-linux/gcc-4.8.3/include/OpenEXR  -O3 -DNDEBUG -fPIC   -Wall -Wno-unused-local-typedefs -Wno-unused-result -fno-math-errno -msse4.2 -std=c++11 -o CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -c /mnt/bonus/dev/rez-build/openimageio/dev/src/src/libutil/argparse.cpp
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -207,7 +207,7 @@ jobs:
 
   vfxplatform-2021-exr3:
     # Test what's anticipated to be VFX Platform 2021 -- mainly, that means
-    # gcc9 and C++17, and also the in-progress openexr/imath 3.0.
+    # gcc9 and C++17, and also OpenEXR/Imath 3.0.
     name: "Linux VFX Platform 2021: gcc9/C++17 py3.7 boost-1.73 exr-3.0 qt-5.15"
     runs-on: ubuntu-latest
     container:
@@ -218,7 +218,7 @@ jobs:
       CMAKE_CXX_STANDARD: 17
       PYTHON_VERSION: 3.7
       USE_SIMD: avx2,f16c
-      OPENEXR_VERSION: RB-3.0
+      OPENEXR_VERSION: v3.0.1
     steps:
       - uses: actions/checkout@v2
       - name: Prepare ccache timestamp
@@ -263,7 +263,7 @@ jobs:
       CXX: g++-7
       CMAKE_CXX_STANDARD: 14
       USE_SIMD: sse4.2
-      OPENEXR_VERSION: v2.4.0
+      OPENEXR_VERSION: v2.4.2
       CMAKE_BUILD_TYPE: Debug
     steps:
       - uses: actions/checkout@v2
@@ -307,7 +307,7 @@ jobs:
       CXX: g++-8
       CMAKE_CXX_STANDARD: 17
       USE_SIMD: avx
-      OPENEXR_VERSION: v2.5.3
+      OPENEXR_VERSION: v2.5.5
       PYBIND11_VERSION: v2.5.0
     steps:
       - uses: actions/checkout@v2
@@ -399,7 +399,7 @@ jobs:
       LIBRAW_VERSION: 0.20.2
       LIBTIFF_VERSION: v4.2.0
       OPENCOLORIO_VERSION: v2.0.0
-      OPENEXR_VERSION: v3.0.0-beta
+      OPENEXR_VERSION: v3.0.1
       PUGIXML_VERSION: v1.11.4
       PYBIND11_VERSION: v2.6.2
       PYTHON_VERSION: 3.8
@@ -717,7 +717,7 @@ jobs:
 
   linux-clang:
     # Test compiling with clang on Linux.
-    name: "Linux clang10: clang10 C++14 avx2 exr2.4"
+    name: "Linux clang10: clang10 C++14 avx2 exr2.5"
     runs-on: ubuntu-18.04
     container:
       image: aswf/ci-osl:2021
@@ -765,7 +765,7 @@ jobs:
 
   linux-static:
     # Test building static libs.
-    name: "Linux static libs: gcc7 C++14 exr2.4"
+    name: "Linux static libs: gcc7 C++14 exr2.2"
     runs-on: ubuntu-18.04
     env:
       CXX: g++-7

--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -93,13 +93,13 @@ checked_find_package (OpenEXR REQUIRED
                       VERSION_MIN 2.0
                       RECOMMEND_MIN 2.2
                       RECOMMEND_MIN_REASON "for DWA compression"
-                      PRINT IMATH_INCLUDES)
+                      PRINT IMATH_INCLUDES OPENEXR_INCLUDES)
 # Force Imath includes to be before everything else to ensure that we have
 # the right Imath/OpenEXR version, not some older version in the system
 # library. This shoudn't be necessary, except for the common case of people
 # building against Imath/OpenEXR 3.x when there is still a system-level
 # install version of 2.x.
-include_directories(BEFORE ${IMATH_INCLUDES})
+include_directories(BEFORE ${IMATH_INCLUDES} ${OPENEXR_INCLUDES})
 if (CMAKE_COMPILER_IS_CLANG AND OPENEXR_VERSION VERSION_LESS 2.3)
     # clang C++ >= 11 doesn't like 'register' keyword in old exr headers
     add_compile_options (-Wno-deprecated-register)

--- a/src/cmake/modules/FindOpenEXR.cmake
+++ b/src/cmake/modules/FindOpenEXR.cmake
@@ -70,7 +70,8 @@ if (TARGET OpenEXR::OpenEXR AND TARGET Imath::Imath)
         list (APPEND ILMBASE_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
     endif ()
 
-elseif (TARGET OpenEXR::IlmImf AND TARGET IlmBase::Imath AND OPENEXR_VERSION VERSION_GREATER_EQUAL 2.4)
+elseif (TARGET OpenEXR::IlmImf AND TARGET IlmBase::Imath AND
+        (OPENEXR_VERSION VERSION_GREATER_EQUAL 2.4 OR OpenEXR_VERSION VERSION_GREATER_EQUAL 2.4))
     # OpenEXR 2.4 or 2.5 with exported config
     if (NOT OpenEXR_FIND_QUIETLY)
         message (STATUS "Found CONFIG for OpenEXR 2 (OPENEXR_VERSION=${OpenEXR_VERSION})")
@@ -97,6 +98,12 @@ elseif (TARGET OpenEXR::IlmImf AND TARGET IlmBase::Imath AND OPENEXR_VERSION VER
     if (CMAKE_USE_PTHREADS_INIT)
         list (APPEND ILMBASE_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
     endif ()
+
+    # Correct for how old OpenEXR config exports set the directory one
+    # level lower than we prefer it.
+    string(REGEX REPLACE "include/OpenEXR$" "include" ILMBASE_INCLUDES "${ILMBASE_INCLUDES}")
+    string(REGEX REPLACE "include/OpenEXR$" "include" IMATH_INCLUDES "${IMATH_INCLUDES}")
+    string(REGEX REPLACE "include/OpenEXR$" "include" OPENEXR_INCLUDES "${OPENEXR_INCLUDES}")
 
 else ()
     # OpenEXR 2.x older versions without a config or whose configs we don't

